### PR TITLE
Add a piggy back migration for jpeg turbo

### PR DESCRIPTION
**Pull request by hmaarrfk**
Piggy backs off https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4091

xref: https://github.com/conda-forge/conda-forge.github.io/issues/673

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Hmmm, not sure what is wrong.

**Comment by hmaarrfk** [comment]
@jschueller @ocefpaf I think this is the last piece of the puzzle. If you wanted to help that would be appreciated. I know you were interested in this in the past.

**Comment by eddy-geek** [comment]
I'm just passing by, but isn't "JpegTuroboMigrator" a typo?

**Comment by hmaarrfk** [comment]
> I'm just passing by, but isn't "JpegTuroboMigrator" a typo?

Yes! Definitely a bug!


Sadly, not the one holding me up :(

**Comment by ocefpaf** [comment]
> I'm just passing by, but isn't "JpegTuroboMigrator" a typo?

If I'm being honest. I kind of like TuroboMigrator ;-p

@hmaarrfk thanks for doing this! We should add this to the next meeting agenda and, hopefully, address one of the oldest standing issue in conda-forge.

**Comment by hmaarrfk** [comment]
If you can fix the bug. You can call it whatever you want!

In the same light, if you can fix the bug, i don't think there will be many objections too ING forward with this.

**Comment by beckermr** [comment]
I can try and look at this tonight. Sometimes you need to run the tests locally with a lot of output to see what's happening.

**Comment by hmaarrfk** [comment]
Thank you. Yeah. I mostly tried to copy our qt mini migrator. I used that to troubleshoot alot.

**Comment by beckermr** [comment]
Double check the recipe yaml for after the migration. I don't see jpeg turbo anywhere in it. I also see a potential issue with openjpeg be changed too and I am not sure if you want that.

File: conda_forge_tick/migrators/jpegturbo.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from conda_forge_tick.migrators.core import MiniMigrator
+import os
+
+
+def _parse_jpeg(lines):
+    new_lines = []
+    for line in lines:
+        line = line.replace(" jpeg\n", " libjpeg-turbo\n")
+        line = line.replace(" jpeg", " libjpeg-turbo ")
```
Comment (hmaarrfk) [inline conda_forge_tick/migrators/jpegturbo.py:9]: I think these lines address the concern with `openjpeg`

i am trying to find a "complicated" recipe to test against. So openjpeg is a good thing to include in the final test.

Though maybe stepping stone is required.

File: tests/test_yaml/jpegturbo_pillow_after_meta.yaml [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  host:
+    - python
+    - pip
+    - setuptools
+    # libraries, c.f. https://github.com/python-pillow/Pillow/blob/9.0.0/setup.py#L30-L38
+    - freetype
+    # fribidi and libimagequant disabled due to
+    # GPL discussion, see #109 and #111
+    # - fribidi
+    - jpeg
```
Comment (beckermr) [inline tests/test_yaml/jpegturbo_pillow_after_meta.yaml:39]:
```suggestion
    - libjpeg-turbo
```

**Comment by hmaarrfk** [comment]
wow!!! many thanks

**Comment by beckermr** [comment]
Happy to help! The issue was that the test bumped pillow to a version that didn't appear to exist. The tests run the version update live and so they failed due to the missing version.

**Comment by hmaarrfk** [comment]
um. did you really mean to merge this first? THis is really part of a larger plan. We should merge the other things I listed in https://github.com/conda-forge/conda-forge.github.io/issues/673#issuecomment-1426881229 first right?

**Comment by beckermr** [comment]
Woops!  I thought it was ready. My apologies. I will back it out.

**Comment by hmaarrfk** [comment]
the patch is ready. i think the decision is not yet finalized

**Pull request merged**